AppFramework/Assertion/GREYAssertions.m

```diff
@@ -39,14 +39,17 @@ @implementation GREYAssertions
   GREYCheckBlockWithError assertionBlock = ^BOOL(id element, __strong NSError **errorOrNil) {
     GREYStringDescription *mismatch = [[GREYStringDescription alloc] init];
     if (![matcher matches:element describingMismatchTo:mismatch]) {
-      NSMutableString *reason = [[NSMutableString alloc] init];
       if (!element) {
-        [reason appendFormat:@"Assertion with Matcher failed: No UI element was matched.\n"];
-        [reason appendFormat:@"Matcher: \n%@", [matcher description]];
+        NSString *reason = [NSString stringWithFormat:
```

Comment:
```
Exception Name: com.google.earlgrey.ElementInteractionErrorDomain
Exception Reason: 
Matched UI Element: <UITableViewLabel:0x7ff259d28540; isAccessible=Y; AX.label='Basic Views'; AX.frame={{20, 308}, {374, 44}}; AX.activationPoint={207, 330}; AX.traits='UIAccessibilityTraitStaticText'; AX.focused='N'; frame={{20, 0}, {374, 44}}; opaque; alpha=1; UIE=N; text='Basic Views'>
Did not match: isNil
In assertion matcher: isNil
Exception Details: An assertion failed.
Exception with Assertion: {
  "Assertion Criteria" : "assertWithMatcher:isNil",
  "Element Matcher" : "((kindOfClass('UILabel') || kindOfClass('UITextField') || kindOfClass('UITextView')) && hasText('Basic Views'))"
}
```

Do you think we can format the `Exception Reason:` better? I don't like how it's all in one text blob.

We do not have test cases for the assertion description / info so can we add them ErrorHandlingTest?

Reply:
New output:
```Exception Name: com.google.earlgrey.ElementInteractionErrorDomain
Exception Reason: The following elements do not match the element matcher: isNil 
  1. <UIWindow:0x7f7fac711270; isAccessible=N; AX.frame={{0, 0}, {375, 812}}; AX.activationPoint={187.5, 406}; AX.traits='UIAccessibilityTraitNone'; AX.focused='N'; frame={{0, 0}, {375, 812}}; opaque; alpha=1> 
  2. isNil 

Exception Details: An assertion failed.
Exception with Assertion: {
  "Assertion Criteria" : "assertWithMatcher:isNil",
  "Element Matcher" : "(kindOfClass('UIWindow') && keyWindow)"
}```

Reply:
Can we have 

Element: <UIWindow..
Did not match: isNil ?

Reply:
Then what about the Element Matcher?

Reply:
Resolved: 
```
Exception Reason: Element does not fulfill assertion criteria: isNil 
Element: <UIWindow:0x7fc1f6c13940; isAccessible=N; AX.frame={{0, 0}, {375, 812}}; AX.activationPoint={187.5, 406}; AX.traits='UIAccessibilityTraitNone'; AX.focused='N'; frame={{0, 0}, {375, 812}}; opaque; alpha=1> 
Mismatch: isNil 

Exception Details: An assertion failed.
Exception with Assertion: {
  "Assertion Criteria" : "assertWithMatcher:isNil",
  "Element Matcher" : "(kindOfClass('UIWindow') && keyWindow)"
}
```